packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -417,7 +417,7 @@ class AppDomain extends Domain {
         );
         _sendAppEvent(app, 'stop');
       } catch (error) {
-        _sendAppEvent(app, 'stop', <String, dynamic>{'error': error.toString()});
+        _sendAppEvent(app, 'stop', <String, dynamic>{'error': _toJsonable(error)});
```

Comment:
Ah, I see those other places now. Instead of the structured error result, what if we update `_toJsonable()` for `ToolExit` to be:

```
if (obj is ToolExit) return obj.message:
```

that way the `error` field in the various places are consistently strings?